### PR TITLE
community: Use environment variables to configure healthcheck server

### PR DESCRIPTION
**Pull request by LanceEa**
## Description

This pulls in the work from the community contribution on making the healthcheck server configurable via Environment variables so that we can support IPV6 setups. This will include the PR run and just cleaning up the docs.

See existing community PR.
https://github.com/emissary-ingress/emissary/pull/4301

## Related Issues
Fixes https://github.com/emissary-ingress/emissary/issues/4274

## Testing
CI and this is being used in production for community members product.

## Checklist

- [ ] **Does my change need to be backported to a previous release?**
- [x] **I made sure to update `CHANGELOG.md`.**
- [x] **This is unlikely to impact how Ambassador performs at scale.**
- [x] **My change is adequately tested.**
- [ ] **I updated `DEVELOPING.md` with any any special dev tricks I had to use to work on this code efficiently.**
- [x] **The changes in this PR have been reviewed for security concerns and adherence to security best practices.**

### CODE REVIEW

**Comment by LanceEa** [comment]
@TimonOmsk - FYI...due to the current state of our CI, it requires a maintainer to run it. Due to this is makes it a really slow back and forth process if I was to ask you to make changes. Therefore,  I have gone ahead and pulled in your commit, made some of my suggested changes (added unit tests, consolidated functions, doc comments) but overall the essence of your PR is still the same. 

Once this has been reviewed we will get it landed. Thanks again for the contribution and patience as we continue to evolve and improve our processes.

File: cmd/entrypoint/env.go
Comment (haq204) [inline cmd/entrypoint/env.go:331]: I feel like we should log a warning if a non supported value is set

**Review by haq204 — COMMENTED** [review]
one minor comment but otherwise lgtm

**Review by haq204 — APPROVED** [review]
approving since I don't consider my comment serious enough to block

File: cmd/entrypoint/env.go
Comment (LanceEa) [reply cmd/entrypoint/env.go:331]: I don't want to muddy this simple function with introducing `ctx` so we can log. I think a trade-off would be to add a log line in the handler where it is passed to `net.Listen` so that it is explicit to the user.

**Comment by LanceEa** [comment]
FYI... verifying in https://github.com/datawire/apro/pull/3161 in Edge Stack.

**Review by AliceProxy — APPROVED** [review]

**Pull request merged**
